components/sup/src/manager/service/supervisor.rs

```diff
@@ -167,14 +169,47 @@ impl Supervisor {
     }
 
     #[cfg(windows)]
-    fn user_info(&self, pkg: &Pkg) -> Result<UserInfo> {
+    fn user_info(&self, pkg: &Pkg, launcher: &LauncherCli) -> Result<UserInfo> {
+        // We have changed the implementation of get_current_username in core
+        // to use a win32 call GetUserNameW instead of the USERNAME environment
+        // variable. This introduces a problem if we are using an older launcher
+        // because both might get the current user name and an older launcher would
+        // get a different value if the current user is the SYSTEM user. The API
+        // call returns 'system' but the environment variable would be the host
+        // name followed by a dollar sign. So when an older launcher attempts to
+        // spawn a service and the supervisor is telling it to use the 'system' user
+        // name, the launcher will try to spawn the service using CreateProcessAsUserW
+        // instead of CreateProcessW thinking it needs to spawn the process as a
+        // different user. CreateProcessAsUserW will fail because it will attempt to
+        // authenticate the system user to create its access token which cannot be done
+        // with the system user.
+        //
+        // So we will check the version of the launcher. This version check is added
+        // in the same set of commits that changed the current username inspection
+        // implementation. So if the version handler is not supported, which it would
+        // not be in an older launcher, we will convert the user name to the old style
+        // system user using the host name.
+        let user = {
+            if pkg.svc_user == "system" {
+                match launcher.version() {
+                    Ok(_) => pkg.svc_user.clone(),
```

Comment:
Could we put a match guard here checking the version is greater than what we expect? I know current user change was added at the same time as the launcher version call, but it seems like it would be good to be explicit and would help make the code more self documenting.

Reply:
I had thought about that too. The trouble is I was not sure what version number to use. However now that I think of this, it really doesn't matter since there will be no version at all until this is in dev so I guess I can just use the latest unstable and it will be fine.

Reply:
good call